src/coreclr/tools/Common/CommandLineHelpers.cs

```diff
@@ -234,25 +256,30 @@ public static void MakeReproPackage(string makeReproPath, string outputFilePath,
                         writer.WriteLine(s);
                 }
 
-                string ConvertFromInputPathToReproPackagePath(string inputPath)
+                string ConvertFromOriginalPathToReproPackagePath(bool input, string originalPath)
                 {
-                    if (inputToReproPackageFileName.TryGetValue(inputPath, out string reproPackagePath))
+                    var originalToReproPackageFileName = input ? inputToReproPackageFileName : outputToReproPackageFileName;
+                    if (originalToReproPackageFileName.TryGetValue(originalPath, out string reproPackagePath))
                     {
                         return reproPackagePath;
                     }
 
                     try
                     {
-                        string inputFileDir = inputToReproPackageFileName.Count.ToString();
-                        reproPackagePath = Path.Combine(inputFileDir, Path.GetFileName(inputPath));
-                        archive.CreateEntryFromFile(inputPath, reproPackagePath);
-                        inputToReproPackageFileName.Add(inputPath, reproPackagePath);
+                        string prefix = input ? "_in_" : "_out_";
+                        string reproFileDir = prefix + originalToReproPackageFileName.Count.ToString() + Path.DirectorySeparatorChar;
```

Comment:
Anyone relying on the old format will break. Not sure how big a deal it is, but maybe worth a discussion.

Reply:
I agree this can be a breaking change, although I am not sure what would be the use case of relying on the locally created names of the directories in the repro package. @MichalStrehovsky what do you think?

I have added the prefixes for the convenience reasons to distinguish between input and output dirs.
We could keep input dirs without prefixes and add prefixes only for outputs (since they are a new "feature")

Reply:
I don't think this is a problem - if we think the new layout is better, then I would go with it. While I agree that technically it's breaking, I don't think it's important enough. For example we consider even the command line parameters of `ilc` to be "internal implementation detail", and thus we can change them without worrying too much about breaking things. This is even less visible than that.